src/app/app.module.ts

```diff
@@ -45,6 +46,6 @@ import { FormsModule } from '@angular/forms';
     BrowserAnimationsModule,
   ],
   bootstrap: [AppComponent],
-  providers: [TodoStoreService, InputButtonComponent],
+  providers: [InputButtonComponent, provideComponentStore(TodoStoreService)],
```

Comment:
Interesting... I didn't know you had to call a function to get those lifecycle hooks to work... good catch!